control.lua

```diff
@@ -276,19 +276,20 @@ local function find_note( ent )
     -- note: find_entity doesn't seem to work when the collision box is 0
     local x = ent.position.x
     local y = ent.position.y
-    local entities = ent.surface.find_entities_filtered{name="invis-note",area={{x-0.01,y-0.01},{x+0.01, y+0.01}}}
-    local invis_note
-    for _, entity in ipairs(entities) do
-        if entity.position.x==x and entity.position.y==y then
-            invis_note = entity
-            break
-        end
-    end
+    local invis_note = ent.surface.find_entities_filtered{name="invis-note",area={{x-0.01,y-0.01},{x+0.01, y+0.01}}}[1]
+	game.print(serpent.line(invis_note, {comment=false}))
+    --local invis_note
+    -- for _, entity in ipairs(entities) do
+    --     if entity.position.x==x and entity.position.y==y then
+    --         invis_note = entity
+    --         break
+    --     end
+    -- end
```

Comment:
Honestly we should probably encode the target entity name in the note too, otherwise it's impossible to avoid sticking the note on trees or other entities that happen to be in the same place.